[RDY] vim-patch:8.0.{1514,1519},8.1.1360

janlazo opened the pull request:
**vim-patch:8.0.1514: getting the list of changes is not easy**
Problem:    Getting the list of changes is not easy.
Solution:   Add the getchangelist() function. (Yegappan Lakshmanan,
            closes vim/vim#2634)
vim/vim@07ad816

**vim-patch:8.0.1519: getchangelist() does not use argument as bufname()**
Problem:    Getchangelist() does not use argument as bufname().
Solution:   Use get_buf_tv(). (Yegappan Lakshmanan, closes vim/vim#2641)
vim/vim@341a64c

**vim-patch:8.1.1360: buffer left 'nomodifiable' after :substitute**
Problem:    Buffer left 'nomodifiable' after :substitute. (Ingo Karkat)
Solution:   Save the value of 'modifiable' earlier' (Christian Brabandt,
            closes vim/vim#4403)
vim/vim@80341bc

Added 8.1.1360 to followup on https://github.com/neovim/neovim/pull/10028. I missed a line change from 8.1.1345.

janlazo committed:
vim-patch:8.0.1514: getting the list of changes is not easy

Problem:    Getting the list of changes is not easy.
Solution:   Add the getchangelist() function. (Yegappan Lakshmanan,
            closes vim/vim#2634)
https://github.com/vim/vim/commit/07ad816525da67cab3c0db21d1286d221dbc7477

janlazo committed:
vim-patch:8.0.1519: getchangelist() does not use argument as bufname()

Problem:    Getchangelist() does not use argument as bufname().
Solution:   Use get_buf_tv(). (Yegappan Lakshmanan, closes vim/vim#2641)
https://github.com/vim/vim/commit/341a64c9cabff08e4a7dc8cd932a598e12134457

janlazo commented on src/nvim/eval.c:
```diff
@@ -9547,6 +9547,40 @@ static void f_getbufvar(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+static void f_getchangelist(typval_T *argvars, typval_T *rettv, FunPtr fptr)
+{
+  tv_list_alloc_ret(rettv, 2);
+  (void)tv_get_number(&argvars[0]);  // issue errmsg if type error
```
Is this actually required for Neovim to issue an error message? I left it WIP because of this and the compiler warning from gcc for an unused return value.

justinmk replied:
Very weird.  I guess that's why `emsg_off++`  is needed after.

Can assign it to the `vim_ignored` global, to avoid the failed CI.

If the tests pass then let's leave it at that :)

janlazo committed:
vim-patch:8.1.1360: buffer left 'nomodifiable' after :substitute

Problem:    Buffer left 'nomodifiable' after :substitute. (Ingo Karkat)
Solution:   Save the value of 'modifiable' earlier' (Christian Brabandt,
            closes vim/vim#4403)
https://github.com/vim/vim/commit/80341bcd89764d96f87859a3aac8bc00aad1d762

justinmk commented:
LGTM, still WIP?
